services/ui-src/src/components/reports/ModalDrawerReportPage.tsx

```diff
@@ -1,107 +1,61 @@
 import { useContext, useState } from "react";
 // components
-import { Box, Button, Flex, Heading, useDisclosure } from "@chakra-ui/react";
-import { ReportContext, ReportDrawer } from "components";
+import { Box, Button, Heading, useDisclosure } from "@chakra-ui/react";
+import { AddEditEntityModal, EntityCard, ReportContext } from "components";
 // utils
-import { useUser } from "utils";
-import { AnyObject, ModalDrawerReportPageShape, ReportStatus } from "types";
+import { AnyObject, ModalDrawerReportPageShape } from "types";
 
-export const ModalDrawerReportPage = ({ route, setSubmitting }: Props) => {
-  const { isOpen, onClose, onOpen } = useDisclosure();
-  const { report, updateReport } = useContext(ReportContext);
-  const { full_name, state, userIsStateUser, userIsStateRep } =
-    useUser().user ?? {};
+export const ModalDrawerReportPage = ({ route }: Props) => {
+  const { entityType, dashboard, modal } = route;
+  const { report } = useContext(ReportContext);
+  const [selectedEntity, setSelectedEntity] = useState<AnyObject>({});
+  const entities = report?.fieldData[entityType];
 
-  // make state
-  const [currentEntity, setCurrentEntity] = useState<string>("");
-  const [entities, setEntities] = useState<Array<any>>([
-    {
-      title: "Access to hospitals",
-      formJson: {},
-    },
-    {
-      title: "Access to nurse practitioners",
-      formJson: {},
-    },
-    {
-      title: "Access to Primary Care Physicians",
-      formJson: {},
-    },
-  ]);
+  // add/edit entity modal disclosure
+  const {
+    isOpen: addEditEntityModalIsOpen,
+    onOpen: addEditEntityModalOnOpenHandler,
+    onClose: addEditEntityModalOnCloseHandler,
+  } = useDisclosure();
 
-  const openRowDrawer = (entity: string) => {
-    setCurrentEntity(entity);
-    onOpen();
-  };
-
-  const removeEntity = (entityTitle: string) => {
-    setEntities(entities.filter((entity) => entity.title !== entityTitle));
-  };
-
-  const onSubmit = async (formData: AnyObject) => {
-    if (userIsStateUser || userIsStateRep) {
-      setSubmitting(true);
-      const reportKeys = {
-        state: state,
-        id: report?.id,
-      };
-      const dataToWrite = {
-        status: ReportStatus.IN_PROGRESS,
-        lastAlteredBy: full_name,
-        fieldData: formData,
-      };
-      await updateReport(reportKeys, dataToWrite);
-      setSubmitting(false);
+  const openAddEditEntityModal = (entityId?: string) => {
+    if (report && entityId) {
```

Comment:
Why is this checking for an entityId? I had to remove that check to fix a bug with deleting, but curious if that was the wrong fix and I overlooked what it was added for initially

Reply:
ah yeah, that's a holdover from the previous implementation i used as a base. we can remove it for now but Karla will be implementing some version of it in her branch too. removed for now.